core/utils/toolbox.js

```diff
@@ -0,0 +1,158 @@
+/**
```

Comment:
All utilities under /utils are currently generic non-blockly specific utilities. This would change that, just a thought but we should discuss whether this is the right place for this.

Reply:
Yes, things under /utils are supposed to not be blockly specific.

Also, half of this file is types. I know I said the types should be in the same file as the rest of the utils, but what if we moved all the typedefs into another file, and possibly created a directory for typedef-only files?

Reply:
The other issue with that is what would the "namespace" be, one possibility is that we do a ``Blockly.Typedefs`` provide, and require it from `blockly.js`.
I agree with the concern of discoverability, it's more discoverable if in the same file.